### PR TITLE
improved paintbarn

### PR DESCRIPTION
**Pull request by SansPapyrus683**

### CODE REVIEW

File: solutions/usaco-919.mdx [hunk 1]
```diff
@@ -54,47 +59,40 @@ int main() {
...
-	static {
-		try {
-			io = new Kattio("paintbarn");
-		} catch(IOException e) {}
-	}
-
-	static int N, K, A=0;
-	static int[][] psum;
+public class PaintBarn {
+	private static final int WIDTH = 1000;
```
Comment (brad182) [inline solutions/usaco-919.mdx:63]: any reason this is private?

File: solutions/usaco-919.mdx [hunk 2]
```diff
@@ -54,47 +59,40 @@ int main() {
...
-			psum[c][d]++;
-			psum[a][d]--;
-			psum[c][b]--;
+		Kattio io = new Kattio("paintbarn");
+		int rectNum = io.nextInt();
+		int paintReq = io.nextInt();
+		int[][] barn = new int[WIDTH + 1][WIDTH + 1];
+
+		for (int i = 0; i < rectNum; i++) {
+			int start_x = io.nextInt();
```
Comment (brad182) [inline solutions/usaco-919.mdx:71]: ew underscore variable names

Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:71]: legit the convention for c++

File: solutions/usaco-919.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:63]: i keep access levels as limited as possible

Comment (darren-yao) [reply solutions/usaco-919.mdx:63]: I would recommend making the variable public. I think a lot of students probably don't know what the private keyword means in Java, and clarity of code takes precedence over good software engineering practices, in my opinion.

Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:63]: if they don't know what private means why would they know what public means

File: solutions/usaco-919.mdx [hunk 3]
```diff
@@ -103,4 +101,36 @@ public class paintbarn {
...
 
 </JavaSection>
+<PySection>
+
+```py
+WIDTH = 1000
+
+barn = [[0 for _ in range(WIDTH + 1)] for _ in range(WIDTH + 1)]
+with open('paintbarn.in') as read:
+	rect_num, paint_req = [int(i) for i in read.readline().split()]
```
Comment (ryanchou-dev) [inline solutions/usaco-919.mdx:111]:
```suggestion
	rect_num, paint_req = map(int, read.readline().split())
```

File: solutions/usaco-919.mdx [hunk 4]
```diff
@@ -103,4 +101,36 @@ public class paintbarn {
...
+	for _ in range(rect_num):
+		start_x, start_y, end_x, end_y = [int(i) for i in read.readline().split()]
```
Comment (ryanchou-dev) [inline solutions/usaco-919.mdx:113]:
```suggestion
		start_x, start_y, end_x, end_y = map(int, read.readline().split())
```

File: solutions/usaco-919.mdx [hunk 5]
```diff
@@ -2,7 +2,7 @@
 id: usaco-919
 source: USACO Silver 2019 February
 title: Painting the Barn
-author: Qi Wang, Nathan Wang
+author: Qi Wang, Nathan Wang, Kevin Sheng
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_paintbarn_silver_feb19.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-919.mdx:8]:
```suggestion
[Official Analysis (C++)](http://www.usaco.org/current/data/sol_paintbarn_silver_feb19.html)
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
minor fixes

File: solutions/usaco-919.mdx (hunk 1 above)
Comment (brad182) [reply solutions/usaco-919.mdx:63]: well you don't need public or private you just do static int ...

and also you type public static void every single time, you almost never type private in competitive programming

File: solutions/usaco-919.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:111]: just asking, is there any particular reason? i can commit this, sure, but just asking

Comment (ryanchou-dev) [reply solutions/usaco-919.mdx:111]: i think it looks cleaner, but it's up to you if you want to commit it.

**Review by ryanchou-dev — APPROVED** [review]
nice

File: solutions/usaco-919.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:63]: who's "you" now

File: solutions/usaco-919.mdx (hunk 2 above)
Comment (brad182) [reply solutions/usaco-919.mdx:71]: i thought that was only for finals/constants

Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:71]: all the style guides i've read have said to use snake_case

File: solutions/usaco-919.mdx (hunk 1 above)
Comment (brad182) [reply solutions/usaco-919.mdx:63]: it is not necessary to do public or private for global variables but I don't think this is a required change

**Review by brad182 — APPROVED** [review]

File: solutions/usaco-919.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/usaco-919.mdx:63]: yeah imo just remove `private`, it doesn't really serve a purpose and may confuse people newer to java

Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:63]: don't think a simple access modifier would confuse people?

File: solutions/usaco-919.mdx [hunk 6]
```diff
@@ -54,47 +59,40 @@ int main() {
...
-
-			// Setting up the prefix sums array with all the corners of the given rectangle
-			psum[a][b]++;
-			psum[c][d]++;
-			psum[a][d]--;
-			psum[c][b]--;
+		Kattio io = new Kattio("paintbarn");
+		int rectNum = io.nextInt();
+		int paintReq = io.nextInt();
+		int[][] barn = new int[WIDTH + 1][WIDTH + 1];
```
Comment (HarryWangATX) [inline solutions/usaco-919.mdx:68]: why are we doing this? do dynamic sizing

File: solutions/usaco-919.mdx [hunk 7]
```diff
@@ -54,47 +59,40 @@ int main() {
...
-					A++;
+		int valid_area = 0;
+		// Run 2D prefix sums on the array
+		for (int x = 0; x <= WIDTH; x++) {
+			for (int y = 0; y <= WIDTH; y++) {
+				if (x > 0) barn[x][y] += barn[x - 1][y];
+				if (y > 0) barn[x][y] += barn[x][y - 1];
+				if (x > 0 && y > 0) barn[x][y] -= barn[x - 1][y - 1];
+				if (barn[x][y] == paintReq) {
+					valid_area++;
```
Comment (HarryWangATX) [inline solutions/usaco-919.mdx:91]: i like the og one better :eyes:

File: solutions/usaco-919.mdx [hunk 8]
```diff
@@ -19,31 +19,36 @@ From the official analysis (with minor modifications):
...
-			if (psum[i][j] == k) ret++;
+
+	int valid_area = 0;
+	// Run 2D prefix sums on the array
+	for (int x = 0; x < WIDTH; x++) {
+		for (int y = 0; y < WIDTH; y++) {
+			if (x > 0) barn[x][y] += barn[x - 1][y];
+			if (y > 0) barn[x][y] += barn[x][y - 1];
+			if (x > 0 && y > 0) barn[x][y] -= barn[x - 1][y - 1];
+			valid_area += barn[x][y] == paint_req;
```
Comment (HarryWangATX) [inline solutions/usaco-919.mdx:48]: read java comment below

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/usaco-919.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:68]: how- the barn's width is explicitly specified to be 1000

File: solutions/usaco-919.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:91]: why lol

File: solutions/usaco-919.mdx (hunk 8 above)
Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:48]: why lol

File: solutions/usaco-919.mdx (hunk 1 above)
Comment (ar492) [reply solutions/usaco-919.mdx:63]: not too confusing, but better to do what's more common (not write `private`)

**Review by ar492 — APPROVED** [review]
The rest looks good

File: solutions/usaco-919.mdx (hunk 6 above)
Comment (HarryWangATX) [reply solutions/usaco-919.mdx:68]: wait mb i remebered the question wrong

File: solutions/usaco-919.mdx (hunk 7 above)
Comment (HarryWangATX) [reply solutions/usaco-919.mdx:91]: why we doing all these conditional statements when we can one line it :eyes:

Comment (SansPapyrus683) [reply solutions/usaco-919.mdx:91]: no one likes giant one-liners harry

**Pull request merged**